hivemq-edge/src/frontend/src/extensions/datahub/utils/node.utils.ts

```diff
@@ -78,18 +78,41 @@ export const validConnections: ConnectionValidity = {
   [DataHubNodeType.FUNCTION]: [[DataHubNodeType.OPERATION, OperationData.Handle.FUNCTION]],
 }
 
-export const isValidPolicyConnection = (connection: Connection, nodes: Node[]) => {
+export const isValidPolicyConnection = (connection: Connection, nodes: Node[], edges: Edge[]) => {
   const source = nodes.find((node) => node.id === connection.source)
   const destination = nodes.find((node) => node.id === connection.target)
 
-  if (!source) {
+  const hasCycle = (node: Node, visited = new Set()) => {
+    if (visited.has(node.id)) return false
+
+    visited.add(node.id)
+
+    for (const outgoer of getOutgoers(node, nodes, edges)) {
+      if (outgoer.id === connection.source) return true
+      if (hasCycle(outgoer, visited)) return true
+    }
```

Comment:
i would add `return false` at the end

Reply:
Oh nicely spotted! 
I wonder if I should consider using strict true/false for the tests too